### PR TITLE
Fix warnings for PHP 8.1, Deprecations in Dokuwiki 2023-04-04a “Jack Jackrum”

### PR DESCRIPTION
**Pull request by Miro-Collas**
- Fix for #46 
- Fix for #44 with thanks to @nerun

### CODE REVIEW

**Comment by glensc** [comment]
what php version are the fixes for?

**Comment by Miro-Collas** [comment]
> what php version are the fixes for?

8, as noted in the PR title

**Comment by glensc** [comment]
8 has many versions already by now: 8.0, 8.1, 8.2, 8.3, maybe even 8.4

**Comment by nerun** [comment]
If you click the links provided by author: [Undefined array key action.php line 74 (with FIX) #44](https://github.com/glensc/dokuwiki-plugin-pageredirect/issues/44) you will see that fix was tested for `PHP 8.1`. It's always recommended to analyze the changes in the code a PR do too. The undefined array key fix uses `array_pad`, that works since PHP 4, as you can see here: [PHP: array_pad - Manual](https://www.php.net/manual/en/function.array-pad.php). There's no reason to it don't work in ANY PHP 8 version: 8.1, 8.2, 8.3, 8.4...


The other one is a DokuWiki deprecation warning. It is how things works for DokuWiki 2023-04-04a “Jack Jackrum”. You don't need to accept it since a warning is not a bug. But a deprecation warning is an advise that the "old way" will stop working sometime in the future.

**Comment by Miro-Collas** [comment]
> 8 has many versions already by now: 8.0, 8.1, 8.2, 8.3, maybe even 8.4

Yes, 8.1 - thanks @nerun  :-)

**Comment by glensc** [comment]
it's not maintainers' job to click around to find info. should be filled in pr body: pr should be standalone describing it's changes.

File: syntax.php [hunk 1]
```diff
@@ -7,6 +7,8 @@
  * @author  David Lorentsen <zyberdog@quakenet.org>
  */
 
+use dokuwiki\File\PageResolver;
```
[the remark this answers is not in the document]
Comment (glensc) [reply syntax.php:10]: I don't have the energy to figure out this right now, so restored compatibility: c5a70f56a3ef18aa4226dce8d62f38eba64f95a5

**Comment by Miro-Collas** [comment]
Thank you.

**Pull request merged**